### ISSUE TITLE
[Bug]: Accessibility issues with SideNavMenu

### ISSUE THREAD

**Issue by folkforms** [carbon-design-system/carbon#20062]
### Package

@carbon/react

### Browser

Chrome, Edge

### Package version

1.24.0

### React version

18.2.0

### Description

Hi there,

I just wanted to raise a couple of issues I'm seeing with the SideNavMenu component. Based on the document linked below I'm assuming these items are mandatory for WCAG compliance, but if I'm wrong please let me know.

1. Not using headings

https://www.w3.org/WAI/ARIA/apg/patterns/accordion/ - "Each accordion header button is wrapped in an element with role heading that has a value set for aria-level that is appropriate for the information architecture of the page."

From reading the above it seems like there needs to be some sort of wrapper around the button, whether a `h1-6` or an element with role="heading" and level="...".

2. Missing aria-controls

https://www.w3.org/WAI/ARIA/apg/patterns/accordion/ - "The accordion header button element has aria-controls set to the ID of the element containing the accordion panel content."

The header button should have aria-controls="<id>" where `id` is the id of the list it controls.

### Reproduction/example

~https://react.carbondesignsystem.com/?path=/docs/components-accordion--overview~ (wrong link, I am using side nav)

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav

### Steps to reproduce

Inspect the DOM of any of the accordion menu examples.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Gururajj77** [comment]
@mbgower , is this really something that needs to be done as it's already wrapped in a `<nav>` tag?? 

Mentioning here that Carbon's side nav is using unordered lists, and the APG example has group of `<div>` tags.

**Comment by mbgower** [comment]
The Aria Authoring Practices are informative, not normative. They are examples containing techniques and recommendations. That is, you do not need to follow them to meet WCAG (which is a distinct standard from ARIA as well). But they do provide a reality check to prompt for if it makes sense to deviate from them.

Accordions are not necessarily headings, but in my experience, most are.

[Carbon has guidance](https://carbondesignsystem.com/components/accordion/accessibility/#design-recommendations) on designers annotating headings, where appropriate. I would personally prefer Carbon to invert their model, and have accordions titles be set as headings by default, and have designers designate when they _shouldn't_ be headings. I think originally there were concerns with what the default semantic level would be for headings (i.e., if out of the box, every accordion contains an h3 heading, it can mess up the hierarchical page structure), but I think Carbon developed a means where the heading is set relative to its parent? If I'm right about this, then revisiting the decision on _not_ settings headings by default may be worth revisiting.

Using `aria-controls` is a standard implementation for accordions. The [Carbon documentation](https://carbondesignsystem.com/components/accordion/accessibility/#development-considerations) states that that attribute is used and it looks like its present in storybook. So I'm surprised to hear that is not being set. @alisonjoseph is this intentional? If you've found a way to establish a relationship between the accordion parent and its disclosed child that doesn't use `aria-controls`, that should be documented.

But hold on @folkforms ! I'm not sure why you're citing accordion components when the topic of the issue is about side nav menu. A menu normally has an `aria-haspopup` not `aria-controls`

**Comment by folkforms** [comment]
hi @mbgower, I'm using "accordion menu" in the general sense, i.e. the side nav menu is an accordion menu, although it's not using the Carbon Accordion component

**Comment by mbgower** [comment]
Could you please provide a link to the specific component implementation you are referring to, instead of accordion. Thanks!

BTW, a better pattern to consult on the APG is probably https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/

**Comment by folkforms** [comment]
> Could you please provide a link to the specific component implementation you are referring to, instead of accordion. Thanks!

this one: https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav

**Comment by mbgower** [comment]
> Not using headings

This is a hierarchical list of links. The top level links are in a list, with the nested links in a secondary list. A heading is not appropriate.

> Missing aria-controls

I'll ask @tombrunet or @alisonjoseph to weigh in, but I don't think it's necessary to have aria-controls in such a clear hierarchical list experience. I don't think it would be _wrong_ to have it. I'm just not sure what you think is missing here without it. I'll note that there have been [ongoing discussions](https://github.com/w3c/aria-practices/issues/805) about when it is necessary.

**Comment by tombrunet** [comment]
Navigation, unfortunately, seems to live in this weird space where a significant chunk of the population thinks it should be a widget, and another significant chunk thinks it should behave like content. I don't believe the community as a whole has really settled on how these should really operate.

As for headings, I agree with Mike. The APG is focused on Accordion in the usage of content where the accordion is basically to take a large page and cut it down into more manageable sections. If the accordion weren't used, those would be headings, so you should preserve the fact that those would be headings when you introduce the accordion. As for the navigation, if the whole list were visible, you wouldn't use headings, so headings are not appropriate.

As for aria-controls - frankly, it's hard to say because, as far as I know, AT's don't really do anything with that information. Strictly speaking, yes, aria-controls should probably be used here. But, there's also a question of if selecting the item should indicate an aria-controls relationship with the resulting content and how that should be indicated.

Overall, I'd classify this as a bug for cleaner code. Without a known impact to users or regulatory compliance, I can't help make an argument for prioritization.
